kong/plugins/jwt/handler.lua

```diff
@@ -157,8 +157,12 @@ local function do_authentication(conf)
     return error(err)
   end
 
-  local www_authenticate_base = conf.realm and fmt('Bearer realm="%s"', conf.realm) or 'Bearer'
-  local www_authenticate_with_error = www_authenticate_base .. ' error="invalid_token"'
+  local realm = conf.realm and fmt(' realm="%s"', conf.realm) or ""
+  local www_authenticate_base = "Bearer" .. realm
+  local www_authenticate_with_error = www_authenticate_base
+    .. (realm ~= "" and "," or "")
+    .. ' error="invalid_token"'
```

Comment:
I think this looks mostly ok. I'd suggest one change - let's split this logic a little bit. 
```suggestion
  local www_authenticate_base
  local www_authenticate_with_error
  if conf.realm then
    www_authenticate_base = fmt('Bearer realm="%s"', conf.realm)
    www_authenticate_with_error = www_authenticate_base .. ', error="invalid_token"'
  else
    www_authenticate_base = 'Bearer'
    www_authenticate_with_error = www_authenticate_base .. ' error="invalid_token"'    
  end
```

This way we don't have to double check if `conf.realm` exist in order to set `realm` to `""` and then check again if `realm` was set to `""` to determine whether or not use comma.

Reply:
other options:

```
local www_authenticate_base
if conf.realm then
  www_authenticate_base = fmt('Bearer realm="%s",', conf.realm)
else
  www_authenticate_base = 'Bearer'
end
local www_authenticate_with_error = www_authenticate_base .. ' error="invalid_token"'
```
or
```
local www_authenticate_base =
    'Bearer' .. (conf.realm and fmt(' realm="%s",', conf.realm) or '')
local www_authenticate_with_error = www_authenticate_base .. ' error="invalid_token"'
```
all are fine by me.

Reply:
Thank you both @nowNick and @samugi for your suggestions! I appreciate the feedback.
I think @nowNick's suggestion is very clear and addresses the issue directly by separating the logic more cleanly.
Regarding @samugi's options, they look good but there's a small issue: both options would add a comma at the end of `www_authenticate_base` when `conf.realm` is set (e.g., `Bearer realm=\"test\",`). This would be correct when we're adding more parameters after it, but might not be ideal when `www_authenticate_base` is used alone (like in the \"Unauthorized\" response where we don't add the error parameter).
Thanks again for the review!

Reply:
I've updated the code to use the cleaner approach as suggested.

Reply:
oh yeah I didn't realize we were reusing `www_authenticate_base` below - my bad!